ambry-frontend/src/main/java/com/github/ambry/frontend/FrontendRestRequestService.java

```diff
@@ -236,7 +236,8 @@ public void start() throws InstantiationException {
             namedBlobDb, idConverter, router, quotaManager);
     s3PostHandler = new S3PostHandler(s3MultipartUploadHandler);
     s3PutHandler = new S3PutHandler(namedBlobPutHandler, s3MultipartUploadHandler, frontendMetrics);
-    s3ListHandler = new S3ListHandler(namedBlobListHandler, frontendMetrics);
+    s3ListHandler =
+        new S3ListHandler(namedBlobListHandler, securityService, frontendMetrics, accountAndContainerInjector);
```

Comment:
Honestly, I think we should create a new Handler, S3GetObjectLockConfigurationHandler. It would be so much simpler if we have another handler. We will call s3BaseHandler.isGetObjectLockConfigurationRequest in the handleGet method and call this new handler right away. This way, it doesn't mix with S3ListHandler.